config/files/finder.go

```diff
@@ -13,8 +13,16 @@ const (
 var configLocations = [][]string{
 	{"."},
 	{"/etc", "flyte", "config"},
+	{UserHomeDir(), ".flyte"},
 	{os.ExpandEnv("$GOPATH"), "src", "github.com", "lyft", "flytestdlib"},
 }
+var osUserHomDir = os.UserHomeDir
```

Comment:
why a variable? Why not directly use it in the function?

Reply:
I dont know if we should add the home dir directly here. I would prefer that to be done only for flytectl

Reply:
can we not just pass the homeDir location to `FindConfigFiles`.
The reason is - this config loading is general purpose and used across all of Flyte.

Reply:
Variable to help with  unit testing  so that i dont have to rely to the filesystem call.
FakeUserHomeDir function mocks the behavior.

Right now its used by flytectl only so yeah will move it to that repo only.